modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php

```diff
@@ -0,0 +1,1201 @@
+<?php
+
+namespace Drupal\tide_content_collection\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\StringTextareaWidget;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\tide_content_collection\SearchApiIndexHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
+
+/**
+ * Implementation of the content collection configuration widget.
+ *
+ * @FieldWidget(
+ *   id = "content_collection_configuration",
+ *   label = @Translation("Content Collection Configuration"),
+ *   field_types = {
+ *     "content_collection_configuration"
+ *   }
+ * )
+ */
+class ContentCollectionConfigurationWidget extends StringTextareaWidget implements ContainerFactoryPluginInterface {
+
+  /**
+   * The Search API Index helper.
+   *
+   * @var \Drupal\tide_content_collection\SearchApiIndexHelper
```

Comment:
@joshua-salsadigital wrong type. Must use the interface instead of the concrete class.